File: Java/gestion-commande/src/main/java/com/ism/restaurant/model/HistoriqueEtatCommande.java
package com.ism.restaurant.model;

import java.time.LocalDateTime;

public class HistoriqueEtatCommande {
    private int id;
    private EtatCommande ancienEtat;
    private EtatCommande nouvelEtat;
    private LocalDateTime dateChangement;

    public HistoriqueEtatCommande() {}

    public HistoriqueEtatCommande(EtatCommande ancienEtat, EtatCommande nouvelEtat, LocalDateTime dateChangement) {
        this.ancienEtat = ancienEtat;
        this.nouvelEtat = nouvelEtat;
        this.dateChangement = dateChangement;
    }

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }
    public EtatCommande getAncienEtat() { return ancienEtat; }
    public void setAncienEtat(EtatCommande ancienEtat) { this.ancienEtat = ancienEtat; }
    public EtatCommande getNouvelEtat() { return nouvelEtat; }
    public void setNouvelEtat(EtatCommande nouvelEtat) { this.nouvelEtat = nouvelEtat; }
    public LocalDateTime getDateChangement() { return dateChangement; }
    public void setDateChangement(LocalDateTime dateChangement) { this.dateChangement = dateChangement; }

    @Override
    public String toString() {
        return "HistoriqueEtatCommande [id=" + id + ", ancienEtat=" + ancienEtat + ", nouvelEtat=" + nouvelEtat + ", dateChangement=" + dateChangement + "]";
    }
}

File: src/Entity/HistoriqueEtatCommande.php
<?php

namespace App\Entity;

use App\Repository\HistoriqueEtatCommandeRepository;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: HistoriqueEtatCommandeRepository::class)]
#[ORM\Table(name: '`historique_etat_commande`')]
class HistoriqueEtatCommande
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 30, nullable: true)]
    private ?string $ancien_etat = null;

    #[ORM\Column(length: 30)]
    private ?string $nouvel_etat = null;

    #[ORM\Column]
    private ?\DateTime $date_changement = null;

    #[ORM\ManyToOne(inversedBy: 'historiqueEtatCommandes')]
    private ?Commande $commande = null;

    #[ORM\ManyToOne(inversedBy: 'historiqueEtatCommandes')]
    private ?User $gestionnaire = null;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getAncienEtat(): ?string
    {
        return $this->ancien_etat;
    }

    public function setAncienEtat(?string $ancien_etat): static
    {
        $this->ancien_etat = $ancien_etat;

        return $this;
    }

    public function getNouvelEtat(): ?string
    {
        return $this->nouvel_etat;
    }

    public function setNouvelEtat(string $nouvel_etat): static
    {
        $this->nouvel_etat = $nouvel_etat;

        return $this;
    }

    public function getDateChangement(): ?\DateTime
    {
        return $this->date_changement;
    }

    public function setDateChangement(\DateTime $date_changement): static
    {
        $this->date_changement = $date_changement;

        return $this;
    }

    public function getCommande(): ?Commande
    {
        return $this->commande;
    }

    public function setCommande(?Commande $commande): static
    {
        $this->commande = $commande;

        return $this;
    }

    public function getGestionnaire(): ?User
    {
        return $this->gestionnaire;
    }

    public function setGestionnaire(?User $gestionnaire): static
    {
        $this->gestionnaire = $gestionnaire;

        return $this;
    }
}
